src/ServerScriptService/InstanceComponents/Misc/Sprint.lua

```diff
@@ -0,0 +1,50 @@
+local Players = game:GetService("Players")
+local ReplicatedStorage = game:GetService("ReplicatedStorage")
+local Trove = require(ReplicatedStorage.Packages.Trove)
+local Comm = require(ReplicatedStorage.Packages.Comm)
+local WalkSpeedConfig = require(ReplicatedStorage.Common.Config.WalkSpeed)
+
+local Sprint = {}
+Sprint.__index = Sprint
+
+function Sprint.new(character: Model)
+	local serverComm = Comm.ServerComm.new(character, "Sprint")
+	local self = setmetatable({}, Sprint)
+
+	self._comm = serverComm
+	self._trove = Trove.new()
+	self._trove:Add(self._comm)
+	self._owner = Players:GetPlayerFromCharacter(character.Parent)
+	self.Instance = character
+
+	self._trove:Connect(character.AncestryChanged, function()
+		self._owner = Players:GetPlayerFromCharacter(character.Parent)
+	end)
```

Comment:
Is this necessary? I'm not sure I understand exactly what it does.

Reply:
It was necessary for the jetpack, but in this case as long as the Sprint tag is applied to the character it becomes no longer necessary.